client/www/scripts/modules/manager/manager.controllers.js

```diff
@@ -609,6 +616,59 @@ Manager.controller('ManagerMainController', [
       }
     };
 
+    function updateHostEnv(host, env) {
+      var deferred = $q.defer();
+      var command =  {
+        cmd: 'env-set',
+        env: env
+      };
+
+      if (host) {
+        host.action(command, function(err, res) {
+          if (err) {
+            deferred.reject(err);
+          } else {
+            deferred.resolve(env);
+          }
+        });
+      }
```

Comment:
I would simply it even more and handle the error on the caller:
`return host.action(command);`

I don't think the promise will work if you just return `if !host` though.